pkg/util/validation/exporter/ring.go

```diff
@@ -113,17 +130,41 @@ func (c *RingConfig) toRingConfig() ring.Config {
 	return rc
 }
 
+// Validate the Config.
+func (c *RingConfig) Validate() error {
+	if c.WaitStabilityMinDuration > 0 {
+		if c.WaitStabilityMinDuration > c.WaitStabilityMaxDuration {
+			return errors.New("-overrides-exporter.ring.wait-stability-max-duration must be greater or equal " +
+				"to -overrides-exporter.ring.wait-stability-min-duration")
+		}
+	}
+	return nil
+}
+
 // overridesExporterRing is a ring client that overrides-exporters can use to
-// assume ownership of a tenant shard
+// establish a leader replica that is the unique exporter of per-tenant limit metrics.
 type overridesExporterRing struct {
-	config     RingConfig
+	services.Service
+
+	config RingConfig
+
 	client     *ring.Ring
 	lifecycler *ring.BasicLifecycler
+
+	subserviceManager *services.Manager
+	subserviceWatcher *services.FailureWatcher
+	logger            log.Logger
 }
 
-// Owns looks up whether this ring member owns the given tenant.
-func (o *overridesExporterRing) Owns(tenantID string) (bool, error) {
-	return instanceOwnsIdentifier(o.client, o.lifecycler.GetInstanceAddr(), tenantID)
+// isLeader checks whether this instance is the leader replica that exports metrics for all tenants.
+func (r *overridesExporterRing) isLeader() (bool, error) {
```

Comment:
[nit] By convention, we're used to keep the `newXXX()` function right after the struct definition. Please move this function below.

Reply:
👍